client/app/scripts/charts/nodes-chart.js

```diff
@@ -356,7 +353,7 @@ export default class NodesChart extends React.Component {
 
     // inject metrics and save coordinates for restore
     const layoutNodes = graph.nodes
-      .mergeDeep(nodeMetrics)
+      .mergeDeep(stateNodes)
```

Comment:
This needs to be done after the coords have been backed up (swap map and mergeDeep), because state nodes can have circle coords (just try select and deselect a node to see the error).